packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -376,9 +454,29 @@ class BabelPluginGetPluginsFromGatsbyConfig {
               return
             }
 
+            // Only allow module.exports = {}
+            // Not things like module.exports = () => ({})
+            if (!t.isObjectExpression(right)) {
+              return
+            }
+
             const plugins = right.properties.find(p => p.key.name === `plugins`)
 
-            plugins.value.elements.map(node => {
+            let pluginsList = []
+
+            if (t.isCallExpression(plugins.value)) {
+              pluginsList = plugins?.value?.callee?.object?.elements
+            } else {
+              pluginsList = plugins?.value?.elements
+            }
+
+            if (!pluginsList) {
+              throw new Error(
+                `Your gatsby-config.js format is currently not supported by Gatsby Admin. Please share your gatsby-config.js file via the "Send feedback" button. Thanks!`
+              )
+            }
+
+            pluginsList.map(node => {
```

Comment:
What happens later if the plugin list has a `...spread`?

Reply:
Could you clarify this? If the `...spread` happens in gatsby-config.js?

Reply:
Yes. The AST for spread is annoyingly a different kind of node than a regular property node. `{plugins: [...arr]}` (see astexplorer.net)

Reply:
Good to know! For now we won't support it then I guess.